mcs/class/System.Net.Http/CFNetworkHandler.cs

```diff
@@ -173,6 +195,43 @@ CFHTTPMessage CreateWebRequestAsync (HttpRequestMessage request)
 		}
 
 		protected internal override async Task<HttpResponseMessage> SendAsync (HttpRequestMessage request, CancellationToken cancellationToken)
+		{
+			return await SendAsync (request, cancellationToken, true).ConfigureAwait (false);
+		}
+		
+		static async Task<HttpRequestMessage> CreateHttpRequestMessageRedirectAsync (HttpRequestMessage req)
+		{
+			var str = req.RequestUri.OriginalString;
+			var clone = new HttpRequestMessage (req.Method, str);
+
+			// Copy the request's content (via a MemoryStream) into the cloned object
+			var ms = new MemoryStream ();
+			if (req.Content != null) {
+				await req.Content.CopyToAsync (ms).ConfigureAwait (false);
+				clone.Content = new StreamContent (ms);
+
+				// Copy the content headers
+				if (req.Content.Headers != null)
+					foreach (var h in req.Content.Headers)
+						clone.Content.Headers.Add (h.Key, h.Value);
+			}
+
+			clone.Version = req.Version;
+
+			foreach (KeyValuePair<string, object> prop in req.Properties) {
```

Comment:
All this would be much better handled inside HttpRequestMessage. Perhaps just do it there and then remove `Authorization`

Reply:
This is about the properties, but no the headers, am I confused. The ignore of the header can be found in line 226.

Reply:
What I meant is doing the close from inside HttpRequestMessage would be much better because you'd not need to manually clone all individual fields

Reply:
We can reuse the obj. Latests push should fix the issue.